bftengine/src/bftengine/Crypto.cpp

```diff
@@ -58,23 +58,19 @@ namespace impl {
 //#define RSA_STANDARD PKCS1v15
 //#define RSA_STANDARD OAEP<SHA>
 
-static RandomPool sGlobalRandGen;  // not thread-safe !!
-
-void CryptographyWrapper::init(const char* randomSeed) {
-  string s(randomSeed);
-  if (s.length() < 16) s.resize(16, ' ');
-  sGlobalRandGen.IncorporateEntropy((CryptoPP::byte*)s.c_str(), s.length());
-
-  VERIFY(DigestUtil::digestLength() == DIGEST_SIZE);
-
-  // Initialize RELIC library for BLS threshold signatures
-  BLS::Relic::Library::Get();
-}
+// Wrap RandomPool in a struct which initialises it. The wrapper is used as static thread local variable so that each
+// thread has got its own instance. The reason for this complication is thread safety. This RandomPool is used by the
+// SigManager which is a singleton used by many threads concurrently.
+struct RandomPoolWrapper : RandomPool {
+  RandomPoolWrapper() {
+    SecByteBlock seed(32);
+    // this might block for a while - 'man 4 random' for details
+    OS_GenerateRandomBlock(true, seed, seed.size());
+    IncorporateEntropy(seed, seed.size());  // method of RandomPool
+  }
+};
 
-void CryptographyWrapper::init() {
-  std::string seed = IntToString(time(NULL));
-  CryptographyWrapper::init(seed.c_str());
-}
+static thread_local RandomPoolWrapper sGlobalRandGen;
```

Comment:
Nit: The comment about "why" this variable is thread_local should be right here.